Game/MyProject/Game/Gameplay/InfoManager.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <stdio.h>
+#include <IvVector3.h>
+#include <IvVector2.h>
+#include <map>
+#include "../SceneManagement/SceneNode.hpp"
+#include "../Car/CarController.hpp"
+
+//Manager supports more than one player
+
+class InfoManager{
+    
+public:
+    InfoManager(SceneNode* root);
+    ~InfoManager();
+    
+    void Update(float dt);
+    
+private:
+    
+    float sinceStartTimer = 0.0f;
+    
+    //range to trigger next checkpoint
+    float checkpointTriggerDistance = 0.5f;
+    
+    //the root (good to search for objects)
+    SceneNode* root;
+    
+    std::vector<std::shared_ptr<SceneNode>> carsList;
```

Comment:
```suggestion
    std::vector<ceneNode*> carsList;
```
candidate for circular dependency problems.